src/dodal/plans/__init__.py

```diff
@@ -0,0 +1,23 @@
+from .scanspec import scan
+from .wrapped import count
+
+"""
+This package is intended to hold MsgGenerator functions which act as self-contained
+experiments: they start runs, collect data, and close the runs. While they may be used
+as building blocks for larger nested plans, they are primarily intended to be run as-is,
+and any common functionality which may be useful to multiple plans extracted to stubs/.
+
+Plans:
+- Must have type hinted arguments, Should use the loosest sensible bounds
+- Must have docstrings describing behaviour and arguments of the function
+- Must not have variadic args or kwargs, Should pass collections instead
+- Must have optional argument named 'metadata' to add metadata to run(s)
+- Must add 'plan_args' to metadata with complete representation including defaults, None
+- Must add 'detectors', 'motors' metadata with list of names of relevant devices
+- Should pass 'shape' to metadata if the run's shape is knowable
```

Comment:
Should: Can we enforce this in CI?

Reply:
`test_plans_comply` in tests/plans/test_compliance checks


- [ ] Type hinted args, [loosest sensible bounds is judgement call for review]
- [ ] Has doc string [whether they adequately describe behaviour/args is left to reviewer, could parse the docstring str to ensure all args are mentioned if we wanted to]
- [ ] No VarArgs/VarKwargs
- [ ] Must have optional argument named 'metadata' with default None not {}
- [ ] Adds 'plan_args' to metadata with complete representation including defaults, None
- [ ] Must add 'detectors', 'motors' metadata with list of names of relevant devices
- [ ] Should pass 'shape' to metadata if the run's shape is knowable

Reply:
So everything that is knowable just from the signature/method object without trying to pass any arguments and step through it. 
Same for the the Stubs requirements, which is just the first 3 and the 4th as optional behaviour (e.g. not for move_relative, which doesn't call a stub that takes md).